x-pack/plugins/upgrade_assistant/server/lib/reindexing/index_settings.ts

```diff
@@ -86,20 +82,18 @@ const removeUnsettableSettings = (settings: FlatSettings['settings']) =>
     'index.uuid',
     'index.blocks.write',
     'index.creation_date',
-    'index.legacy',
-    'index.mapping.single_type',
     'index.provided_name',
     'index.routing.allocation.initial_recovery._id',
     'index.version.created',
-    'index.version.upgraded',
     'index.verified_before_close',
+    // Below settings are deprecated in 9.0
+    'index.version.upgraded',
   ]);
 
 // Use `flow` to pipe the settings through each function.
 const transformSettings = flow(removeUnsettableSettings);
 
 const updateFixableMappings = (mappings: FlatSettings['mappings']) => {
-  // TODO: change type to _doc
```

Comment:
I don't think this comment is applicable anymore. I've talked to Alan W. and an index with a type defined should still work in 8.0, it will just be ignored.